src/vignette/http/middleware.clj

```diff
@@ -46,7 +47,16 @@
           (header "X-Cache-Hits" "ORIGIN")
           (header "Connection" "close")))))
 
-(defn request-timer [handler]
+(defn request-timer
+  [handler]
   (fn [request]
     (perf/publish {:request-count 1})
     (perf/timing :request-time (handler request))))
+
+(defn bad-request-path
+  []
+  (fn [request]
+    (perf/publish {:bad-request-path-count 1})
+    (log/warn (str "bad-request-path: " (:uri request)))
+    (-> (render "Unrecognized request path!\nSee https://github.com/Wikia/vignette for documentation.\n" request)
```

Comment:
`not-found` is generated using a [macro that adds a wrapper](https://github.com/weavejester/compojure/blob/master/src/compojure/route.clj#L44) that seems to interfere with adding this as middleware. I tried that first but had trouble getting it to work, so I just reproduced the core of the function here.